vm/src/builtins/type.rs

```diff
@@ -405,6 +405,35 @@ impl PyType {
             .unwrap_or_else(|| vm.ctx.new_str(self.name().deref()).into())
     }
 
+    #[pyproperty(magic, setter)]
+    fn set_qualname(&self, value: PyObjectRef, vm: &VirtualMachine) -> PyResult<()> {
+        if value.is(&vm.ctx.none()) {
+            return Err(vm.new_type_error(format!(
+                "cannot delete '{}' attribute of immutable type '{}'",
+                "__qualname__",
+                self.name()
+            )));
+        };
+        if !self.slots.flags.has_feature(PyTypeFlags::HEAPTYPE) {
```

Comment:
This sound to me at now as all static types have IMMUTABLE flag.
But i think we need to add comment about it. We need to replace this HEAPTYPE flag check to IMMUTABLE flag check in future.

Reply:
I added comment as you advised. Thank you